core/src/main/java/org/elasticsearch/index/mapper/core/StringFieldMapper.java

```diff
@@ -155,6 +173,9 @@ public StringFieldMapper build(BuilderContext context) {
                     iterator.remove();
                 } else if (propName.equals("position_offset_gap")) {
                     builder.positionOffsetGap(XContentMapValues.nodeIntegerValue(propNode, -1));
+                    if (builder.positionOffsetGap < 0) {
+                        throw new MapperParsingException("positions_offset_gap less than 0 aren't allowed.");
```

Comment:
I know the exc we throw here effectively discards `builder` but I think it's less smelly to first assign to a local "untrusted" var first, e.g. `int newPositionOffsetGap`, and check whether that's < 0, before calling `builder.positionOffsetGap(newPositionOffsetGap)`.